### PR TITLE
Remove hooks on gradient accumulation on engine/optimizer destroy

### PR DESCRIPTION
**Pull request by chiragjn**
Fixes: #4856 
See: https://github.com/pytorch/pytorch/issues/46386

### CODE REVIEW

**Comment by chiragjn** [comment]
@microsoft-github-policy-service agree

**Comment by chiragjn** [comment]
@mrwyattii requesting for a review

**Review by tjruwase — APPROVED** [review]
